protostar/commands/test/environments/execution_environment.py

```diff
@@ -0,0 +1,56 @@
+from abc import ABC, abstractmethod
```

Comment:
I'd consider moving this class out of  the `commands/test/environments` because it's going to be used by two commands.